common4j/versioning/version_tasks.gradle

```diff
@@ -0,0 +1,78 @@
+def getVersionFile() {
+    
+    File file = new File("$rootDir/common/common4j/versioning/version.properties")
+    if(file.exists()){
+        return file;
+    }else{
+        return new File("$rootDir/common4j/versioning/version.properties")
+    }
+
+}
+
+def getVersionProps() {
+    def versionProps = new Properties();
+    getVersionFile().withInputStream {stream -> versionProps.load(stream)}
+    return versionProps
+}
+
+private String getVersionNamePatch() {
+    return (getVersionProps()['versionName'] =~ /[^.]+/)[2].toString()
+}
+
+private Integer getVersionNameMinor() {
+    return (getVersionProps()['versionName'] =~ /\d+/)[1].toInteger()
+}
+
+private Integer getVersionNameMajor() {
+    return (getVersionProps()['versionName'] =~ /\d+/)[0].toInteger()
+}
+
+private Integer getLatestPatchVersion() {
+    return getVersionProps()['latestPatchVersion'].toInteger()
+}
+
+ext.getAppVersionCode = {
+    getVersionProps()['versionCode'].toInteger()
+}
+
+ext.getAppVersionName = {
+    getVersionProps()['versionName'].toString()
+}
+
+private void saveChanges(String versionName) {
+    def versionProps = getVersionProps()
+    versionProps['versionName'] = versionName
+    versionProps.store(getVersionFile().newWriter(), null)
+}
+
+private void incrementLatestVersion(){
+    def versionProps = getVersionProps()
+    versionProps["latestPatchVersion"] = (getLatestPatchVersion() + 1).toString()
+    versionProps.store(getVersionFile().newWriter(), null )
+}
+
+task incrementLatestVersionNumber(){
+    doLast{
+        incrementLatestVersion()
+    }
+}
+
+task versionSnapshot {
+    doLast {
+        def versionNameMajor = getVersionNameMajor()
+        def versionNameMinor = getVersionNameMinor()
+        def versionNamePatch = getVersionNamePatch()
+        def versionName = "${versionNameMajor}.${versionNameMinor}.${versionNamePatch}-SNAPSHOT".toString()
+        saveChanges(versionName)
+    }
+}
+
+task versionLatest {
+    doLast {
+        def versionNameMajor = '0'
```

Comment:
I'm surprised that this works with '0'.

Shouldn't this be getVersionNameMajor() and getVersionNameMinor()?

Reply:
Latest is always versioned like: `0.0.n`